### PR TITLE
Celery worker liveness probe

### PR DESCRIPTION
**Pull request by KulykDmytro**
### Airflow chart change

Adding celery worker liveness probe to determine whether worker is alive

Based on native `celery ping` command

### CODE REVIEW

File: chart/templates/workers/worker-deployment.yaml [hunk 1]
```diff
@@ -213,6 +213,20 @@ spec:
...
+            failureThreshold: {{ .Values.workers.livenessProbe.failureThreshold }}
+            periodSeconds: {{ .Values.workers.livenessProbe.periodSeconds }}
+            exec:
+              command:
+                {{- if .Values.workers.livenessProbe.command }}
+                {{ toYaml .Values.workers.livenessProbe.command | nindent 16 }}
+                {{- else}}
+                - sh
+                - -c
+                - exec /entrypoint python -m celery --app airflow.executors.celery_executor.app inspect ping -d celery@${HOSTNAME}
```
Comment (jedcunningham) [inline chart/templates/workers/worker-deployment.yaml:228]:
```suggestion
                - exec /entrypoint celery --app airflow.executors.celery_executor.app inspect ping -d celery@${HOSTNAME}
```

Let's do it this way as that's how the celery docs do it.

Comment (jedcunningham) [inline chart/templates/workers/worker-deployment.yaml:228]:
```suggestion
                - CONNECTION_CHECK_MAX_COUNT=0 exec /entrypoint python -m celery --app airflow.executors.celery_executor.app inspect ping -d celery@${HOSTNAME}
```

Let's do this as well to keep the probe quick.

File: chart/values.yaml [hunk 2]
```diff
@@ -475,6 +475,14 @@ workers:
     # of local-path provisioner.
     fixPermissions: false
 
+  # If the celery stops heartbeating for 5 minutes (5*60s) kill the
+  # celery and let Kubernetes restart it
```
Comment (jedcunningham) [inline chart/values.yaml:479]:
```suggestion
  # If the worker stops responding for 5 minutes (5*60s) kill the
  # worker and let Kubernetes restart it
```

nit: it's not really a heartbeat like the scheduler

**Review by jedcunningham — CHANGES_REQUESTED** [review]
Thanks! Can you also add a test like the [one we have for the scheduler liveness probe](https://github.com/apache/airflow/blob/c063fc688cf20c37ed830de5e3dac4a664fd8241/tests/charts/test_scheduler.py#L225)?

**Comment by jedcunningham** [comment]
Hey @KulykDmytro, gentle bump, I'd love to get this in the next chart release.

**Comment by KulykDmytro** [comment]
> Hey @KulykDmytro, gentle bump, I'd love to get this in the next chart release.
performing testing on our end - will update you upon completion

**Comment by anu251989** [comment]
> > Hey @KulykDmytro, gentle bump, I'd love to get this in the next chart release.
> 
> performing testing on our end - will update you upon completion

Do you have any update on this. I tried from my end and getting 

 kubectl exec airflow-worker-0 -n <namespace>   -- celery --app airflow.executors.celery_executor.app inspect ping -d celery@${HOSTNAME}
Error: No nodes replied within time constraint
command terminated with exit code 69

we have to monitor the celery health checks inorder to resume the connection between redis and airflow-worker pods. 
in logs, could see missed heartbeat from worker-2. 
after read so many posts felt it is helpful for me

**Comment by potiuk** [comment]
It's not @jedcunningham - it's @KulykDmytro who worked on it. If he does not want to continue (?) maybe @anu251989 you would like to pick it up and lead it to completion? You could become one of > 2000 contributors to Airflow.

**Pull request closed without merging**